src/Microsoft.Azure.ServiceBus/QueueClient.cs

```diff
@@ -197,7 +197,8 @@ internal SessionPumpHost SessionPumpHost
                             this.sessionPumpHost = new SessionPumpHost(
                                 this.ClientId,
                                 this.ReceiveMode,
-                                this.SessionClient);
+                                this.SessionClient,
+                                this.ServiceBusConnection.Endpoint.ToString());
```

Comment:
Should this be `Endpoint.ToString()` or `Endpoint.Authority`?
Former would give out `amqps://abcd.servicebus.windows.net:80`, and latter would be `abcd.servicebus.windows.net`.
I feel the latter is cleaner. 
This will also relieve the need of #197